etl/churn.kp/knowledge.md

```diff
@@ -507,20 +497,61 @@ def compute_churn_week(df, week_start, bucket, prefix):
                 x.get('is_active', 'unknown')
             ),(
                 1,  # active users 
-                x.get('usage_hours', 0),
-                x.get('squared_usage_hours',0) 
+                x.get('usage_hours', 0.0),
+                x.get('squared_usage_hours', 0.0),
+                x.get('total_uri_count', 0),
+                x.get('unique_domains_count', 0)
             )
         )
     )
+    
+    churn_schema = StructType([
```

Comment:
Can we move this closer to the explanatory prose ("The fields we want in the output are...") above? Or maybe just move the per-field descriptions here as code comments next to each field.